pbf/models.py

```diff
@@ -231,12 +231,35 @@ class Meta:
     #
     # To see the list of spirits that use this field, see spirit_specific_resource_name below.
     spirit_specific_resource = models.IntegerField(default=0)
+    # some spirits allow you to do something to their spirit-specific resource,
+    # but only once per turn.
+    # again, to reduce some repetitive code, we'll use one field for all of these.
+    spirit_specific_per_turn_flags = models.PositiveIntegerField(default=0)
+
+    # Meanings for specific bits in the spirit-specific per-turn flags:
+    # It's okay for different spirits to assign different meanings to the same bits.
+    # But do note that no spirit should assign a different meaning to these generic ones:
+    # (these can be useful for e.g. Unconstrained Sharp Fangs Behind the Leaves)
+    SPIRIT_SPECIFIC_INCREMENTED_THIS_TURN = 1 << 0
+    SPIRIT_SPECIFIC_DECREMENTED_THIS_TURN = 1 << 1
+    # Spirit-specific bits:
+    # Spreading Rot Renews the Earth:
+    ROT_GAINED_THIS_TURN = 1 << 2 # Whether they've gained from their track (NOT incrementing using the +1 button)
+    ROT_CONVERTED_THIS_TURN = 1 << 3
```

Comment:
How many flags can we create this way in a single integer field? I assume the answer is plenty, but alternatively we could be a bit more thoughtful here and have a map from spirits to flags (so 1 means "Rot Gained" for Rot, but "Energy Added to Impending Cards" for Dances).

Reply:
yeah Django guarantees that the field is at least 32 bits, but I fully expect that we should probably just do what you've described here anyway.

Reply:
May be a bit more cumbersome than its worth, since 32 bits is a fair amount of wiggle room. But, if we ever hit the threshold, fixing it after the fact would break the spirits a bit for existing games.

Will leave it up to you if you wanna hack on this further.

Reply:
Well, either way, I think these four flags in this pull request are set for sure, regardless of what happens later. I am planning to have Dances Up Earthquakes use `1 << 2` which I think is sensible: it has no Rot, so the `ROT_` flags hold no meaning, so might as well reuse `1 << 2`. If Dances Up Earthquakes were to start at `1 << 4` and then the next spirit at `1 << 5`, there is a chance we'll run out eventually (but, likelihood of that unknown), so I think this makes sense.

Reply:
Ah yea, you can reuse safely without having a spirit map controlling functionality 👍